### PR TITLE
*: parse more lock options in alter table statement

### PR DESCRIPTION
**Pull request by holys**
Currently, tidb only supports  `LOCK = NONE`, but lack of support for `LOCK = [DEFAULT | SHARED | EXCLUSIVE]`. 

Why need this?

Since TiDB has no support for  multiple schema change, so syncer has to split ddl statements like 
`ALTER TABLE foo.bar ADD COLUMN name varchar(20) NOT NULL, LOCK=DEFAULT`, and syncer use `tidb/parser` to parse SQL into AST statements。 But parser can not recognize `LOCK=DEFAULT` clause,  which make syncer hard to deal with such statement.

@shenli  @zimulala  @hanfei1991   PTAL

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,5 @@ tags
 temp_parser_file
 y.output
 profile.coverprofile
+*.a
+.vscode
```
Comment (GregoryIan) [inline .gitignore]: Why not put this rule into your global .gitignore file?

Comment (holys) [inline .gitignore]: ok

File: parser/parser.y [hunk 2]
```diff
@@ -442,6 +442,7 @@ import (
 	engine		"ENGINE"
 	engines		"ENGINES"
 	escape 		"ESCAPE"
+	exclusive   "EXCLUSIVE"
```
Comment (shenli) [inline parser/parser.y]: Keep alignment.

File: parser/parser.y [hunk 3]
```diff
@@ -483,6 +484,7 @@ import (
 	serializable	"SERIALIZABLE"
 	session		"SESSION"
 	share		"SHARE"
+	shared       "SHARED"
```
Comment (shenli) [inline parser/parser.y]: Alignment.

File: parser/parser.y [hunk 4]
```diff
@@ -657,6 +659,7 @@ import (
 	LoadDataStmt		"Load data statement"
 	LocalOpt		"Local opt"
 	LockTablesStmt		"Lock tables statement"
+	LockClause         "Alter table lock clause"
```
Comment (shenli) [inline parser/parser.y]: Alignment.

**Comment by shenli** [comment]
@zimulala @tiancaiamao  PTAL

File: parser/parser.y (hunk 4 above)
Comment (holys) [inline parser/parser.y]: I don't know how to keep alignment in vscode or sublime text. I have to do it in vim.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by holys** [comment]
@zimulala  PTAL

File: ast/ddl.go [hunk 5]
```diff
@@ -646,6 +646,17 @@ const (
 // TODO: Add more actions
 )
 
+// LockType https://dev.mysql.com/doc/refman/5.7/en/alter-table.html#alter-table-concurrency
+type LockType byte
+
+// Lock Types
```
Comment (zimulala) [inline ast/ddl.go]: Complete this comment.

File: parser/parser.y [hunk 6]
```diff
@@ -1030,13 +1033,31 @@ AlterTableSpec:
...
+LockClause: 
+	"LOCK" eq "NONE"
+	{
+		$$ = ast.LockTypeNone
+	}
+|	"LOCK" eq "DEFAULT"
+	{
+		$$ = ast.LockTypeDefault
+	}
+|   "LOCK" eq "SHARED"
```
Comment (zimulala) [inline parser/parser.y]: Format alignment.

File: parser/parser.y [hunk 7]
```diff
@@ -1030,13 +1033,31 @@ AlterTableSpec:
...
+	{
+		$$ = ast.LockTypeShared
+	}
+|   "LOCK" eq "EXCLUSIVE"
```
Comment (zimulala) [inline parser/parser.y]: Ditto.

File: ast/ddl.go [hunk 8]
```diff
@@ -646,6 +646,17 @@ const (
 // TODO: Add more actions
 )
 
+// LockType https://dev.mysql.com/doc/refman/5.7/en/alter-table.html#alter-table-concurrency
```
Comment (zimulala) [inline ast/ddl.go]: The format of this comment should look like this:
```
LockType ... ...
See https://dev.mysql.com/doc/refman/5.7/en/alter-table.html#alter-table-concurrency
```

**Comment by holys** [comment]
ping @zimulala

**Comment by zimulala** [comment]
LGTM

**Pull request merged**